src/components/Search/index.tsx

```diff
@@ -105,20 +110,43 @@ const handleSearchResult = async (
 const Search: FC<{
   content?: string
   hasButton?: boolean
+  truncateTypeHash?: boolean
   onEditEnd?: () => void
-}> = memo(({ content, hasButton, onEditEnd }) => {
+}> = memo(({ content, hasButton, onEditEnd, truncateTypeHash }) => {
+  const SEARCH_COUNT = 10
   const isMobile = useIsMobile()
   const { t } = useTranslation()
+  const [searchByType, setSearchByType] = useSearchType()
+  const [isSearchByName, setIsSearchByName] = useState(searchByType === 'name')
+  const [searchByNameResults, setSearchByNameResults] = useState<UDTQueryResult[] | null>(null)
   const history = useHistory()
-  const SearchPlaceholder = t('navbar.search_placeholder')
   const [searchValue, setSearchValue] = useState(content || '')
-  const [placeholder, setPlaceholder] = useState(SearchPlaceholder)
   const inputElement = useRef<HTMLInputElement>(null)
 
-  // update input placeholder when language change
-  useEffect(() => {
-    setPlaceholder(SearchPlaceholder)
-  }, [SearchPlaceholder])
+  const toggleSearchType = () => {
+    const newIsSearchByNames = !isSearchByName
+    const searchTypePersistValue = newIsSearchByNames ? 'name' : 'id'
+    setSearchByType(searchTypePersistValue)
+    setIsSearchByName(newIsSearchByNames)
+    setSearchByNameResults(null)
+  }
+
+  const { refetch: refetchSearchByName } = useQuery(
+    ['searchByName', searchValue],
+    () =>
+      explorerService.api.fetchSearchByNameResult(searchValue).then(searchResult => {
+        setSearchByNameResults(searchResult ? searchResult.slice(0, SEARCH_COUNT) : [])
```

Comment:
`searchCount` sounds more like an API-level filtering option.

In this case, where truncation is being done at the UI level, it should be `displayedCount` instead of `searchCount`.

Alternatively, you can consider adding a `size` option to the backend API to control the number of results.

Additionally, you should annotate the reason for truncation.

Reply:
updated

Reply:
> Additionally, you should annotate the reason for truncation.

This is necessary, otherwise the purpose of this code snippet would have multiple possibilities.

Future developers (even yourself after a year) may not fully understand the true intention behind this implementation.

Reply:
added

Reply:
updated